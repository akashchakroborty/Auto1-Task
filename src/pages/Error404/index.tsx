import React from 'react';
import { makeStyles } from '@mui/styles';
import { Container, Typography, Link } from '@mui/material';
import { Link as RouterLink } from 'react-router-dom';
import Logo from '../../assets/logo.png';

const useStyles = makeStyles({
  container: {
    padding: 24,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  image: {
    width: 'min-content',
    marginBottom: 24,
  }
});

const Error404: React.FC = () => {
  const classes = useStyles();
  return (
    <Container className={classes.container} maxWidth="sm">
      <img src={Logo} alt="Auto1 Logo" className={classes.image} />
      <Typography variant="h3" fontWeight={700} gutterBottom>
        404 - Not Found
      </Typography>
      <Typography variant="subtitle1" component="h5" gutterBottom>
        Sorry, the page you are looking for does not exists.
      </Typography>
      <Typography variant="subtitle1" gutterBottom>
        You can always go back to the
        {' '}
        <Link color="primary" component={RouterLink} to="/">
          homepage
        </Link>
        .
      </Typography>
    </Container>
  );
};

export default Error404;
