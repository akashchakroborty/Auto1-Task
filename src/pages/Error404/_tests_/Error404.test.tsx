import React from 'react';
import { shallow } from 'enzyme';
import Error404 from '..';

describe('should test the Error404 Component', () => {
  test('should render the component', () => {
    const component = shallow(<Error404 />);
    expect(component).toMatchSnapshot();
  });
});
