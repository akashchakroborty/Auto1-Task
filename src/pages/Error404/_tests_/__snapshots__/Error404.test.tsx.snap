// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the Error404 Component should render the component 1`] = `
<ForwardRef(Container)
  className="makeStyles-container-1"
  maxWidth="sm"
>
  <img
    alt="Auto1 Logo"
    className="makeStyles-image-2"
    src=""
  />
  <ForwardRef(Typography)
    fontWeight={700}
    gutterBottom={true}
    variant="h3"
  >
    404 - Not Found
  </ForwardRef(Typography)>
  <ForwardRef(Typography)
    component="h5"
    gutterBottom={true}
    variant="subtitle1"
  >
    Sorry, the page you are looking for does not exists.
  </ForwardRef(Typography)>
  <ForwardRef(Typography)
    gutterBottom={true}
    variant="subtitle1"
  >
    You can always go back to the
     
    <ForwardRef(Link)
      color="primary"
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/"
    >
      homepage
    </ForwardRef(Link)>
    .
  </ForwardRef(Typography)>
</ForwardRef(Container)>
`;
